lib/Echidna/Processor.hs

```diff
@@ -107,30 +107,18 @@ instance FromJSON SlitherInfo where
         v <- o .: "value"
         t <- o .: "type"
         case t of
-          'u':'i':'n':'t':x ->
-            case AbiUInt <$> readMaybe x <*> readMaybe v of
-              Nothing -> failure v t
-              i -> pure i
-
-          'i':'n':'t':x ->
-            case AbiInt <$> readMaybe x <*> readMaybe v of
-              Nothing -> failure v t
-              i -> pure i
-
+          'u':'i':'n':'t':x -> pure $ AbiUInt <$> readMaybe x <*> readMaybe v
+          'i':'n':'t':x -> pure $ AbiInt <$> readMaybe x <*> readMaybe v
           "string" ->
             pure . Just . AbiString $
               if "0x" `isPrefixOf` v
               then fromRight (error ("invalid b16 decoding of: " ++ show v)) $ BS16.decode $ BSU.fromString $ drop 2 v
               else BSU.fromString v
 
-          "address" ->
-            case AbiAddress . Addr <$> readMaybe v of
-              Nothing -> failure v t
-              a -> pure a
+          "address" -> pure $ AbiAddress . Addr <$> readMaybe v
 
           -- we don't need all the types for now
           _ -> pure Nothing
-        where failure v t = fail $ "failed to parse " ++ t ++ ": " ++ v
```

Comment:
I think that was made on purpose to give a better error message. Is it no longer needed?

Reply:
We want to do this parsing bulletproof, the reason is simple: slither sometimes output types are user defined, for constants, which can break the parsing. If missing a few constants should not break the tool.